evm/forge-test/CircleIntegration.t.sol

```diff
@@ -4,91 +4,1215 @@ pragma solidity ^0.8.13;
+    using BytesLib for bytes;
+
+    bytes32 constant GOVERNANCE_MODULE = 0x000000000000000000000000000000436972636c65496e746567726174696f6e;
+
+    uint8 constant GOVERNANCE_UPDATE_WORMHOLE_FINALITY = 1;
+    uint8 constant GOVERNANCE_REGISTER_EMITTER_AND_DOMAIN = 2;
+    uint8 constant GOVERNANCE_REGISTER_ACCEPTED_TOKEN = 3;
+    uint8 constant GOVERNANCE_REGISTER_TARGET_CHAIN_TOKEN = 4;
+    uint8 constant GOVERNANCE_UPGRADE_CONTRACT = 5;
+
     // USDC
     IUSDC usdc;
 
     // dependencies
+    WormholeSimulator wormholeSimulator;
     IWormhole wormhole;
+
     ICircleBridge circleBridge;
     IMessageTransmitter messageTransmitter;
 
-    // USDC Mint/Burn contracts
-    CircleIntegrationSetup setup;
-    CircleIntegrationImplementation implementation;
-    CircleIntegrationProxy proxy;
+    CircleIntegrationSimulator circleSimulator;
     ICircleIntegration circleIntegration;
 
-    function mintUSDC() public {
-        // spoof .configureMinter() call with the master minter account
-        vm.prank(usdc.masterMinter());
+    // foreign
+    bytes32 foreignUsdc;
+
+    function maxUSDCAmountToMint() public view returns (uint256) {
+        return type(uint256).max - usdc.totalSupply();
+    }
 
+    function mintUSDC(uint256 amount) public {
+        require(amount <= maxUSDCAmountToMint(), "total supply overflow");
+        usdc.mint(address(this), amount);
+    }
+
+    function setupWormhole() public {
+        // Set up this chain's Wormhole
+        wormholeSimulator = new WormholeSimulator(
+            vm.envAddress("TESTING_WORMHOLE_ADDRESS"), uint256(vm.envBytes32("TESTING_DEVNET_GUARDIAN")));
+        wormhole = wormholeSimulator.wormhole();
+    }
+
+    function setupUSDC() public {
+        usdc = IUSDC(vm.envAddress("TESTING_USDC_TOKEN_ADDRESS"));
+
+        (, bytes memory queriedDecimals) = address(usdc).staticcall(abi.encodeWithSignature("decimals()"));
+        uint8 decimals = abi.decode(queriedDecimals, (uint8));
+        require(decimals == 6, "wrong USDC");
+
+        // spoof .configureMinter() call with the master minter account
         // allow this test contract to mint USDC
+        vm.prank(usdc.masterMinter());
         usdc.configureMinter(address(this), type(uint256).max);
 
-        // mint $1000 USDC to the test contract (or an external user)
-        usdc.mint(address(this), 1000e6);
+        uint256 amount = 42069;
+        mintUSDC(amount);
+        require(usdc.balanceOf(address(this)) == amount);
     }
 
-    function makeDependencies() public {
-        usdc = IUSDC(vm.envAddress("TESTING_USDC_TOKEN_ADDRESS"));
-        wormhole = IWormhole(vm.envAddress("TESTING_WORMHOLE_ADDRESS"));
-        circleBridge = ICircleBridge(vm.envAddress("TESTING_CIRCLE_BRIDGE_ADDRESS"));
-        messageTransmitter = IMessageTransmitter(vm.envAddress("TESTING_MESSAGE_TRANSMITTER_ADDRESS"));
+    function setupCircleIntegration() public {
+        circleSimulator = new CircleIntegrationSimulator(
```

Comment:
Is this used anymore?

Reply:
Removed the `CircleIntegrationSimulator` contract in 6d7df26c67c115f3de69acd3b1bb0d756a9a7ece.